contract.md

```diff
@@ -12,11 +16,19 @@ like to work on, as part of your project (for example, other things related to D
 you have come across, or topics from other classes)? Why are you interested in working on
 these things?_
 
+As I want to build a syntactically lighweight DSL, my plan is to write it as an external DSL, as an internal DSL would be too limiting and syntactically complex for my vision. With this in mind, I'd like to work on my parsing and interpreting skills, specifically as it pertains to external DSLs. I'm interested in these skills because I see them as applicable across a wide variety of host programming languages, while internal DSL techniques can often be more specialized. 
+
+I'd also like to get some experience designing a DSL in such a way that it could be expanded upon by a capable developer. In the readings for this class we saw different perspectives on how expandable a DSL should be by design. At this point, I've settled on designing my DSL to enable limited future expansion with the addition of new commands (control flow), but not new data structures. Similarly to the language design, I'll have to think carefully about how best to write my backend code with limited modularity in mind. This is a skill I want to develop because these kinds of design decisions come up in lots of CS projects, from hobbyist to professional-grade.
```

Comment:
This would be a very good skill to work on, but it also seems like it would involve spending time streamlining and polishing the backend- which would not be good if it ended up taking away time from polishing the design, or front end implementation. There's nothing wrong with it, though!